payments-ui-core/src/main/java/com/stripe/android/ui/core/analytics/PaymentsUiEventReporterCompose.kt

```diff
@@ -0,0 +1,45 @@
+package com.stripe.android.ui.core.analytics
+
+import androidx.annotation.RestrictTo
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.compositionLocalOf
+import com.stripe.android.core.Logger
+import com.stripe.android.uicore.BuildConfig
+import com.stripe.android.uicore.analytics.LocalUiEventReporter
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+val LocalPaymentsUiEventReporter = compositionLocalOf<PaymentsUiEventReporter> {
+    EmptyPaymentsUiEventReporter
+}
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+@Composable
+fun ReportablePaymentsUi(
+    eventReporter: PaymentsUiEventReporter,
+    content: @Composable () -> Unit
+) {
+    CompositionLocalProvider(
+        LocalPaymentsUiEventReporter provides eventReporter,
+        LocalUiEventReporter provides eventReporter
+    ) {
+        content()
+    }
+}
+
+private object EmptyPaymentsUiEventReporter : PaymentsUiEventReporter {
+    private val logger: Logger
+        get() = Logger.getInstance(BuildConfig.DEBUG)
+
+    override fun onFieldInteracted() {
+        logger.debug("PaymentsUiEventReporter.onFieldInteracted() event not reported")
```

Comment:
It feels like we should crash the app when we're in a debug build here.

Reply:
I worry this will cause End-to-End tests & instrumentation tests to fail because they won't have the `EventsReporter` applied. I could log them as error logs rather than debug logs?

Reply:
It probably will, which means we have places this isn't applied, and we should fix it! Crashing in debug mode seems like the right thing to do.